aviatrix/resource_aviatrix_distributed_Firewalling_zero_trust_rule.go

```diff
@@ -0,0 +1,151 @@
+package aviatrix
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/AviatrixSystems/terraform-provider-aviatrix/v3/goaviatrix"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAviatrixDistributedFirewallingZeroTrustRule() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceAviatrixDistributedFirewallingZeroTrustRuleCreate,
+		UpdateWithoutTimeout: resourceAviatrixDistributedFirewallingZeroTrustRuleUpdate,
+		ReadWithoutTimeout:   resourceAviatrixDistributedFirewallingZeroTrustRuleRead,
+		DeleteWithoutTimeout: resourceAviatrixDistributedFirewallingZeroTrustRuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"action": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"DENY", "PERMIT"}, true),
+				Description: "Action for the specified source and destination Smart Groups." +
+					"Must be one of PERMIT or DENY.",
+			},
+			"logging": {
+				Type:        schema.TypeBool,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Boolean value to enable or disable logging for the zero trust rule.",
+			},
+		},
+	}
+}
+
+func resourceAviatrixDistributedFirewallingZeroTrustRuleUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, ok := meta.(*goaviatrix.Client)
+	if !ok {
+		return diag.Errorf("failed to assert meta as *goaviatrix.Client")
+	}
+
+	action, ok := d.Get("action").(string)
+	if !ok {
+		return diag.Errorf("failed to assert 'action' as string")
+	}
+
+	logging, ok := d.Get("logging").(bool)
+	if !ok {
+		return diag.Errorf("failed to assert 'logging' as bool")
+	}
+
+	zeroTrustRuleConfig := &goaviatrix.DistributedFirewallingZeroTrustRule{
+		Action:  action,
+		Logging: logging,
+	}
+
+	if err := client.UpdateDistributedFirewallingZeroTrust(ctx, zeroTrustRuleConfig); err != nil {
+		return diag.Errorf("failed to update the zero trust rule: %v", err)
+	}
+
+	d.SetId(strings.Replace(client.ControllerIP, ".", "-", -1))
+	return resourceAviatrixDistributedFirewallingZeroTrustRuleRead(ctx, d, meta)
+}
+
+func resourceAviatrixDistributedFirewallingZeroTrustRuleCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, ok := meta.(*goaviatrix.Client)
+	if !ok {
+		return diag.Errorf("failed to assert meta as *goaviatrix.Client")
+	}
+
+	action, ok := d.Get("action").(string)
```

Comment:
Lots of the code in the provider just does a type assertion without the error handling, which seems OK to me since if it failed it would mean the code is broken pretty badly.

Probably ok if you keep this too, if other code is doing the same pattern.